xml/Microsoft.Extensions.Diagnostics.Enrichment/IEnrichmentTagCollector.xml

```diff
@@ -41,10 +43,21 @@
         <Parameter Name="tagValue" Type="System.Object" />
       </Parameters>
       <Docs>
-        <param name="tagName">To be added.</param>
-        <param name="tagValue">To be added.</param>
-        <summary>To be added.</summary>
-        <remarks>To be added.</remarks>
+        <param name="tagName">Enrichment property key.</param>
+        <param name="tagValue">Enrichment property value.</param>
+        <summary>
+            Adds a tag in form of a key/value pair.
+            </summary>
+        <remarks>
+            For log enrichment, <paramref name="tagValue" /> is serialized as per the rules below:
+            <list type="bullet"><listheader><term>Primitive types</term><description>recognized and efficiently serialized.</description></listheader><item><term>Arrays</term><description>recognized and serialized in a loop.</description></item><item><term><see cref="T:System.Collections.IDictionary" /></term><description>recognized as IDictionary&lt;string, object&gt; and serialized in a loop.</description></item><item><term><see cref="T:System.DateTime" /></term><description>recognized and serialized after converting to <see cref="M:System.DateTime.ToUniversalTime" />.</description></item><item><term>All the rest</term><description>converted to <see cref="T:System.String" /> as is and serialized.</description></item></list>
```

Comment:
https://review.learn.microsoft.com/en-us/dotnet/api/microsoft.extensions.diagnostics.enrichment.ienrichmenttagcollector.add?view=dotnet-plat-ext-8.0&branch=pr-en-us-9500#remarks - Need to log a platform bug on this.

Reply:
Tag is described here: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/recommended-tags#list

Reply:
@kexugit @huangmin-ms Is the `<list>` done correctly here? If so, I'll log a bug, since it renders without any space or punctuation between the term and definition (see image). If not, let me know what needs to be changed. The docs aren't clear how to create a "definition list".

<img width="450" alt="image" src="https://github.com/dotnet/dotnet-api-docs/assets/24882762/7b89ea3a-5f35-4014-95b1-99c1a319270b">

Reply:
@gewarren  Yes, please log a bug. I can see the one which works has a new line for each description.
 https://github.com/Azure/azure-docs-sdk-dotnet/blob/073400a2e785231b2dfcd06dd4f0cef48ea0544d/xml/Microsoft.ServiceFabric.Data/RestoreContext.xml#L109

https://learn.microsoft.com/en-us/dotnet/api/microsoft.servicefabric.data.restorecontext.restoreasync?view=azure-dotnet

Reply:
Yes, I think we should add a hyphen followed by a space character.